community/server/src/main/java/org/neo4j/server/configuration/ConfigWrappingConfiguration.java

```diff
@@ -60,31 +60,15 @@ public boolean containsKey( String key )
     public Object getProperty( String key )
     {
         Setting<?> setting = getSettingForKey( key );
-        if( setting == null )
-        {
-            return null;
-        }
-        else
-        {
-            return config.get( setting );
-        }
+        return setting == null ? config.getParams().get( key ) : config.get( setting );
     }
 
     @Override
     public Iterator<String> getKeys()
     {
-        // get all the properties keys
-        Set<String> staticKeys = getRegisteredSettings().keySet();
-        // only keep the properties which have been assigned default values or user-defined values
-        Set<String> notNullKeys = new HashSet<>();
-        for ( String key : staticKeys )
-        {
-            if ( containsKey( key ) )
-            {
-                notNullKeys.add( key );
-            }
-        }
-        return notNullKeys.iterator();
+        Set<String> propertyKeys = new HashSet<>(config.getParams().keySet());
+        propertyKeys.addAll( getRegisteredSettings().keySet() );
```

Comment:
I feel this changes its previous behaviour. Before, as indicated by the comments, this method will only keep the keys of `properties which have been assigned default values` and the key of properties defined by users.